### PR TITLE
use random_int() instead of rand()

### PR DESCRIPTION
**Pull request by alexfinnarn**
My IDE says this is a recommended function change but also suggests handling the exception too...so maybe not the best for an intro tutorial.

### CODE REVIEW

File: knpu/ep1-stellar/javascript-api.md [hunk 1]
```diff
@@ -28,15 +28,15 @@ We want this API endpoint to return JSON... and remember: the *only* rule for
 a Symfony controller is that it must return a Symfony Response object. So we could
 literally say `return new Response(json_encode(['hearts' => 5]))`.
 
-But that's too much work! Instead say `return new JsonResponse(['hearts' => rand(5, 100)]`:
+But that's too much work! Instead say `return new JsonResponse(['hearts' => random_int(5, 100)]`:
```
Comment (bocharsky-bw) [inline knpu/ep1-stellar/javascript-api.md]: Please, revert this to match the video

**Comment by alexfinnarn** [comment]
Thanks for the prompt reply!

I only saw this notice via the "Php Inspections (EA Extended)" plugin to PHPStorm. On the project page, there is an issue about the reason why they included that suggestion. https://github.com/kalessil/phpinspectionsea/issues/59#issuecomment-264636225

Essentially, I think they argue that it is better to have a default of a cryptographically secure value and only change that back to `rand()` when performance is an issue. In that comment, even though the difference in performance can be great when done many, many times, if it is only done a few times, then the difference is negligible. The key is what you do by default, and I wasn't thinking at all about security. I think there is an argument that if PHP devs use more secure functions by default PHP apps, in general, will have fewer potential issues. 

I think a fun note on `mt_rand()` and `random_int()` with links to the PHP manual pages might be something to add. The reason I like the Php Inspections (EA Extended) plugin so much is that it exposes me to things about PHP I never would consider in the first place. 

I'm not sure what the "the ***TIP block" is, but that's about all the info I have on why my IDE told me to make the switch. Hope it helps!

**Comment by alexfinnarn** [comment]
I just saw your note about `rand()` being an alias of `rand()` so I guess there is no difference between the two.

**Comment by bocharsky-bw** [comment]
Hey Alex,

Thanks for your digging on it! I think this function is good to be mentioned 👍, I'll probably just tweak your message a bit. About `***TIP` syntax - it's our internal syntax for notes we add to our courses, they are styled on the front-end with a nice TIP block.

Thank you for your idea one more time!

**Pull request merged**